### PR TITLE
Manually export .d.ts files

### LINKED ISSUE #458: Switch to manual generation of DTS files
Related to #352 and #416. Currently, the Durable SDK auto-generates its `.d.ts` from the source-code `.ts` files. We should switch to manually authoring our type declarations and our `.d.ts` files. This has multiple benefits:

* We can be intentional about which types we export, and what public APIs we support, helping us become more thoughtful about any breaking changes
* Helps us limit the number of unnecessary exports (see #352), avoiding issues such as #353.

### PR DESCRIPTION
**Pull request by hossam-nasr**
Resolves #458. To make reviewing easier, I've split this PR into smaller commits with descriptive names. The first commit does nothing but move files from the `src/types` to the `types` directory and updates imports.

### CODE REVIEW

File: types/orchestrationTypes.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { FunctionOptions, FunctionTrigger, InvocationContext, LogHandler } from "@azure/functions";
-import { Task, TimerTask, TokenSource } from ".";
-import { EntityId, RetryOptions } from "../classes";
+import { RetryOptions, Task, TimerTask, TokenSource } from ".";
+import { EntityId } from "./entityTypes";
```
Comment (ejizba) [inline types/orchestrationTypes.d.ts]: While we're renaming, I'd prefer if we get rid of `Types` from these file names. Now that its in the `types/` folder and has the `.d.ts` extension, I think `Types` is implied

**Review by ejizba — APPROVED** [review]
I like the smaller commits, thanks!

I'm still a bit nervous about the scope of this change, but I think it's worth the risk. I'm glad it's being done now instead of closer to the preview announcement, and I hope we do plenty of manual testing once this is merged. Tbh I expect some bumps, but I think we should be able to handle them

**Review by davidmrdavid — APPROVED** [review]
LGTM. Thanks for the well-split commits.
A follow-up action item would be to ensure that our autogenerated docs [here](https://learn.microsoft.com/en-us/javascript/api/durable-functions/?view=azure-node-latest) continue working after all these changes. Definitely not a preview blocker, but I'd like to finish this validation before GA.

File: types/orchestrationTypes.d.ts (hunk 1 above)
Comment (hossam-nasr) [reply types/orchestrationTypes.d.ts]: Makes sense, done.

**Pull request merged**